react/features/filmstrip/actions.web.js

```diff
@@ -80,21 +85,65 @@ export function setVerticalViewDimensions() {
     return (dispatch: Dispatch<any>, getState: Function) => {
         const state = getState();
         const { clientHeight = 0, clientWidth = 0 } = state['features/base/responsive-ui'];
+        const { width: filmstripWidth } = state['features/filmstrip'];
         const disableSelfView = shouldHideSelfView(state);
-        const thumbnails = calculateThumbnailSizeForVerticalView(clientWidth);
+        const resizableFilmstrip = isFilmstripResizable(state);
+        const _verticalViewGrid = resizableFilmstrip && ((filmstripWidth.current ?? 0) > FILMSTRIP_GRID_BREAKPOINT);
+
+        let gridView = {};
+        let thumbnails = {};
+        let filmstripDimensions = {};
+
+        // grid view in the vertical filmstrip
+        if (_verticalViewGrid) {
+            const dimensions = getTileViewGridDimensions(state, filmstripWidth.current);
+            const {
+                height,
+                width
+            } = calculateThumbnailSizeForTileView({
+                ...dimensions,
+                clientWidth: filmstripWidth.current,
+                clientHeight,
+                disableResponsiveTiles: false,
+                disableTileEnlargement: false,
+                isVerticalFilmstrip: true
+            });
+            const { columns, rows } = dimensions;
+            const thumbnailsTotalHeight = rows * (TILE_VERTICAL_MARGIN + height);
+            const hasScroll = clientHeight < thumbnailsTotalHeight;
+            const widthOfFilmstrip = (columns * (TILE_HORIZONTAL_MARGIN + width)) + (hasScroll ? SCROLL_SIZE : 0);
+            const filmstripHeight = Math.min(clientHeight, thumbnailsTotalHeight);
+
+            gridView = {
+                gridDimensions: dimensions,
+                thumbnailSize: {
+                    height,
+                    width
+                }
+            };
+
+            filmstripDimensions = {
+                height: filmstripHeight,
+                width: widthOfFilmstrip
+            };
+        } else {
+            thumbnails = resizableFilmstrip
+                ? calculateThumbnailSizeForResizableVerticalView(clientWidth, filmstripWidth.current)
+                : calculateThumbnailSizeForVerticalView(clientWidth);
+        }
 
         dispatch({
             type: SET_VERTICAL_VIEW_DIMENSIONS,
             dimensions: {
                 ...thumbnails,
-                remoteVideosContainer: {
+                remoteVideosContainer: _verticalViewGrid ? filmstripDimensions : {
                     width: thumbnails?.local?.width
-                        + TILE_HORIZONTAL_MARGIN + STAGE_VIEW_THUMBNAIL_HORIZONTAL_BORDER + SCROLL_SIZE,
+                        + (TILE_HORIZONTAL_MARGIN / 2) + SCROLL_SIZE,
```

Comment:
Why do we take 1/2 of `TILE_HORIZONTAL_MARGIN`?

Reply:
In order to reduce the space between the tiles and the scrollbar